subprojects/dependency-management/src/main/java/org/gradle/internal/locking/LockFileReaderWriter.java

```diff
@@ -177,6 +177,9 @@ private Path getUniqueLockfilePath() {
 
     private void parseLine(String line, Map<String, List<String>> result) {
         String[] split = line.split("=");
+        if (split.length != 2) {
+            throw new InvalidLockFileException("project '" + context.getProjectPath().getPath() + "'. Line: " + line, null);
```

Comment:
❌ You will need to support more cases than this one. `context.getProjectPath()` can be `null` if the locking is not for a project.

Dependency locking has 3 scopes:
* Gradle settings dependencies
* Project buildscript dependencies
* Project dependencies

I would replace the content of the message with an indication of the file that's broken, by providing its path. You should also leverage the `DocumentationRegistry` as done for `LIMITATIONS_DOC_LINK` to have a link to the documentation explaining the format.